test/e2e/kubernetes/pod/pod.go

```diff
@@ -250,29 +250,25 @@ func RunCommandMultipleTimes(podRunnerCmd podRunnerCmd, image, name, command str
 		podName := fmt.Sprintf("%s-%d", name, r.Intn(99999))
 		var p *Pod
 		var err error
-		p, err = podRunnerCmd(image, podName, "default", command, true, sleep, duration, timeout)
-
+		p, err = podRunnerCmd(image, podName, "default", command, true, sleep, 10*time.Second, timeout)
 		if err != nil {
 			return successfulAttempts, err
 		}
-		succeeded, err := p.WaitOnSucceeded(sleep, duration)
+		succeeded, err := p.WaitOnSucceeded(sleep, timeout)
 		if err != nil {
-			log.Printf("pod did not succeed in time %s\n", err)
+			log.Printf("pod %s did not succeed in time\n", podName)
+			return successfulAttempts, err
 		}
-		cmd := exec.Command("k", "logs", podName, "-n", "default")
-		out, err := cmd.CombinedOutput()
+		terminated, err := p.WaitOnTerminated(podName, sleep, commandTimeout, timeout)
```

Comment:
There is some overlap here (anecdotally Terminated container state strongly correlates with Succeeded pod state), but we do this as a follow-up step to introspect the container execution time to validate that its duration is less than the passed in `commandTimeout`)